pkg/auth/authenticator/password/ldappassword/util/server.go

```diff
@@ -0,0 +1,84 @@
+package ldaputil
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+
+	"github.com/go-ldap/ldap"
+)
+
+// New returns a new LDAPServerConfig
+func New(url LDAPURL, insecure bool, tlsConfig *tls.Config, timeLimit int) LDAPServerConfig {
```

Comment:
doing group syncs, sure. a query that can return 10,000 items I'd probably let run a lot longer than a single user entry lookup